crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -60,6 +65,83 @@ pub trait Struct: Reflect {
     fn clone_dynamic(&self) -> DynamicStruct;
 }
 
+/// A container for compile-time struct info
+#[derive(Clone, Debug)]
+pub struct StructInfo {
+    type_name: Cow<'static, str>,
+    type_id: TypeId,
+    fields: Vec<NamedField>,
```

Comment:
This is immutable upon construction. Capacity and the ability to extend/add fields after construction is unnecessary. Please switch to using `Box<[NamedField]>` instead. Alternatively, if it's possible to make the inputs during construction `const`, a `&'static [NamedField]`.

Reply:
I'll look into the latter first!